tests/Unit/Http/Controllers/Admin/UserControllerTest.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+namespace Tests\Unit\Http\Controllers\Admin;
+
+use App\Http\Controllers\Admin\UserController;
+use App\Http\Requests\UserRequest;
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Repositories\User\UserRepositoryInterface;
+use Mockery as m;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Tests\TestCase;
+
+class UserControllerTest extends TestCase
+{
+    protected $userRepo;
+    protected $userController;
+    protected $userAuthorized;
+    protected $userUnAuthorized;
+    protected $id;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->userRepo = m::mock(UserRepositoryInterface::class)->makePartial();
+        $this->userController = new UserController($this->userRepo);
+        $this->user = [
+            'name' => 'Unauthorized',
+            'email' => 'Unauthorized@gmail.com',
+            'password' => '123456',
+            'address' => 'Long Biên',
+            'phone' => '012345678',
+            'role_id' => 10,
+            'times' => 0,
+            'status' => 0,
+        ];
+        $this->userAuthorized = factory(User::class)->make();
+        $this->userUnAuthorized = new User([
+            'name' => 'Unauthorized',
+            'email' => 'Unauthorized@gmail.com',
+            'password' => '123456',
+            'address' => 'Long Biên',
+            'phone' => '012345678',
+            'role_id' => 10,
+            'times' => 0,
+            'status' => 0,
+        ]);
+        $this->id = 1;
+    }
+
+    public function tearDown(): void
+    {
+        m::close();
+        unset($this->userRepo);
+        unset($this->userAuthorized);
+        unset($this->userUnAuthorized);
+        unset($this->userController);
+        unset($this->user);
+        unset($this->id);
+        parent::tearDown();
+    }
+
+    public function test_index_user_authorized_view()
+    {
+        $this->be($this->userAuthorized);
+        $this->userRepo->shouldReceive('getRole');
```

Comment:
```suggestion
        $this->userRepo->shouldReceive('getRole')
            ->with(...)
            ->once()/twice()/v.v...
            ->andReturn(...);
```
tương tự bên dưới nhé.
chi tiết vào ;)

Reply:
Được ông